src/main/java/pl/simpleascoding/tutoringplatform/domain/user/Role.java

```diff
@@ -22,4 +23,17 @@ public class Role {
     public Role(RoleType role) {
         this.roleType = role;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Role role = (Role) o;
+        return roleType == role.roleType;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(roleType);
+    }
```

Comment:
I think it's better to use Lombok's `@EqualsAndHashCode`.

Reply:
As we talked at discord, it could be problematic in the future